### PR TITLE
Block Patterns List: Fix visual title and tooltip inconsistencies

### LINKED ISSUE #64632: Patterns in the Site Editor Design panel should have visible names
## What problem does this address?

I was writing testing instructions and I wanted to ask the tester / user to change the header template part to use the "Centered header" from the Design panel. But because the patterns do not have visible names, only aria-labels, the tester (who does not use assistive technology) would need to guess which pattern to insert.

Also, the notice in the bottom left corner only says that the header template part is updated, not which pattern that was inserted.

## What is your proposed solution?

- Show a visible tooltip with the name when a user hovers over/focuses on a pattern
- Or always show the name below the pattern.

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes #64632
Related to #64166, #60160

## What?

This PR resolves inconsistencies and introduces clearer rules for the title and tooltip in the `BlockPatternList` component.

## Why?

I think there are three problems with this component currently:

1. Even though the pattern has a visual title and an aria-label applied, a tooltip is rendered.

image

2. In the Design panel for replacing template content, there is no visible title. Additionally, there is no tooltip displayed when we focus on a pattern, so to find out what the pattern's name is, we'll need to use a screen reader or go into the source code and look for aria-label.

image

3. An empty div may be rendered even though no visual title is present.

empty-div

## How?

In this PR, I apply the following rules wherever the `BlockPatternsList` component is used to resolve these inconsistencies:

- In inserters and sidebars, Don't show the visual title. However, user-defined patterns always show the visual title.
- In modals, always show the visual title.
- If the pattern has a visual title, don't show a tooltip. If not, show a tooltip.

To achieve this, I removed the ~~showTitlesAsTooltip~~ `showTitle` prop from the `BlockPatternsList` component. Whether or not to show a tooltip is determined solely by whether or not to show the visual title, i.e. the ~~showTitle~~ `showTitleAsTooltip` prop.

I am also aware that in some places the visual title is intentionally hidden. One example is #60160. Respecting this approach, the visual titles remain unchanged in this PR, with one exception.

The only visual change in this PR is the modal after creating a custom template:

| Before | After |
|--------|--------|
| image| image| 

This is the result based on the rule "In modals, always show the visual title", but we can change the title "Fallback content" or make an exception and not display a visual title.

## Testing Instructions

Check where the `BlockPatternsList` component is used and make sure there are no visual changes.
Make sure that visual titles and tooltips are displayed based on the following rules:

> - In inserters and sidebars, Don't show the visual title. However, user-defined patterns always show the visual title.
> - In modals, always show the visual title.
> - If the pattern has a visual title, don't show a tooltip. If not, show a tooltip.

<details><summary>Pattern inserter</summary>

image

</details> 

<details><summary>Pattern inserter (mobile)</summary>

image

</details> 

<details><summary>Template Parts Design panel</summary>

image

</details> 

<details><summary>Explore all patterns modal</summary>

image

</details> 

<details><summary>Template Parts replace modal</summary>

image

</details> 

<details><summary>Main inserter > Search pattern</summary>

image

</details> 

<details><summary>Query Loop block > Choose a pattern modal</summary>

image

</details> 

<details><summary>Swap Template modal</summary>

image

</details> 

<details><summary>After creating a custom template</summary>

image

</details> 

<details><summary>After creating a new page</summary>

image

</details> 

<details><summary>Transform a Query Loop block to a pattern</summary>

image

</details>

### CODE REVIEW

File: packages/block-editor/src/components/block-patterns-list/README.md [hunk 1]
```diff
@@ -69,6 +69,14 @@ The aria label for the block patterns list.
 -   Required: No
 -   Default: `Block Patterns`
 
+#### showTitle
+
+Whether to show the block pattern title. User-defined patterns always show a title regardless of this prop.
+
+-   Type: `boolean`
+-   Required: No
+-   Default: true
```
Comment (t-hamano) [inline packages/block-editor/src/components/block-patterns-list/README.md]: Add missing documentation

**Comment by richtabor** [comment]
> with one exception.

Why this exception?

**Comment by t-hamano** [comment]
> > with one exception.
> Why this exception?

In this PR, I introduced a rule called "In modals, always show the visual title". As a result, it means that only the modal after creating a custom template did not follow this rule.

> The only visual change in this PR is the modal after creating a custom template:
> 
> This is the result based on the rule "In modals, always show the visual title", but we can change the title "Fallback content" or make an exception and not display a visual title.

**Comment by ntsekouras** [comment]
Thanks for the PR! I think what complicated things a bit more is the `showTitle` prop, since we already had `showTooltip` prop, that was introduced [here](https://github.com/WordPress/gutenberg/pull/46419/). In general the `showTooltip` would handle to show either the title or the tooltip (these are mutually exclusive).

I didn't check thoroughly right now, but I'm wondering if the `showTitle` prop was needed in the first place ([introduced here](https://github.com/WordPress/gutenberg/pull/60160)).

I'm all for consistency, but should we keep the initial `showTooltip` and have all the use cases we want covered? I'm advocating for that prop, because `showTitle` is a bit ambiguous, since we always want to show a title, either visually rendered or as a tooltip in hover.

**Comment by t-hamano** [comment]
Sorry for the late reply.

> should we keep the initial `showTooltip` and have all the use cases we want covered? I'm advocating for that prop, because `showTitle` is a bit ambiguous, since we always want to show a title, either visually rendered or as a tooltip in hover. 

That's true. I've removed the `showTitle` prop and better documented the purpose of the `showTitleAsTooltip` prop.

BTW, this PR will also improve the Change Design popover:

image

**Review by ntsekouras — APPROVED** [review]
LGTM, thank you!

I find it a bit weird that user patterns have always the title displayed, but I don't have the context for that decision and is not related to this PR.

**Pull request merged**